File: med/python/Question1493.py
from typing import List


class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        ans = left = 0
        flag = 0
        for right, num in enumerate(nums):
            if num == 0:
                flag += 1
            while flag == 2:
                if nums[left] == 0:
                    flag -= 1
                left += 1
            ans = max(ans, right - left)
        return ans


print(Solution().longestSubarray([1, 1, 1]))

File: cpp/med/Question1493.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int longestSubarray(vector<int> &nums) {
        int max_len = 0, left = 0;
        bool flag = false;
        for (int right = 0; right < nums.size(); right++) {
            if (flag && nums[right] == 0) {
                while (flag) {
                    if (nums[left] == 0) {
                        left++;
                        break;
                    }
                    left++;
                }
            } else if (nums[right] == 0) {
                flag = true;
            }
            max_len = max(max_len, right - left);
        }
        return max_len;
    }
};
